test/end2end_test.go

```diff
@@ -3901,6 +3901,27 @@ func (s) TestClientRequestBodyErrorCloseAfterLength(t *testing.T) {
 	}
 }
 
+// Tests gRPC server's behavior when a gRPC client sends a frame with an invalid
+// streamID. Per [HTTP/2 spec]: Streams initiated by a client MUST use
+// odd-numbered stream identifiers. This test sets up a test server and send a
+// header frame with stream ID of 2. The test asserts that a subsequent read on
+// the transport throws an error.
+//
+// [HTTP/2 spec]: https://httpwg.org/specs/rfc7540.html#StreamIdentifiers
+func (s) TestClientInvalidStreamID(t *testing.T) {
+	te := newTest(t, tcpClearEnv)
+	te.startServer(nil)
+	defer te.tearDown()
```

Comment:
I didnt like using `newTest()` because it requires you to pass in a `env` parameter. The `test` struct is already overloaded, and let's not reuse that again. At some time in the future, we would love to split this end2end_test package. 

Also we dont need a server and a client here. The setup should essentially be server and serverTester.

I'm thinking about something like this
```suggestion
	lis, err := net.Listen("tcp", "localhost:0")
	if err != nil {
		t.Fatalf("Failed to listen: %v", err)
	}
	defer lis.Close()
	s := grpc.NewServer()
	defer s.Stop()
	go s.Serve(lis)
	conn, err := net.DialTimeout("tcp", lis.Addr().String(), defaultTestTimeout)
	if err != nil {
		t.Fatalf("Failed to dial: %v", err)
	}
	st := newServerTesterFromConn(t, conn)
	st.greet()
```

Reply:
Also, would be nice to have st.greet() really be part of newServerTesterFromConn.. Not imp for this PR